src/brahma/interceptor.cpp

```diff
@@ -11,9 +11,14 @@ size_t total_apis = 0;
 extern int brahma_bind_functions() {
   BRAHMA_LOGGER_INIT();
   if (brahma::bindings == nullptr) {
-#ifdef BRAHMA_ENABLE_MPI
+#if defined(BRAHMA_ENABLE_HDF5) && defined(BRAHMA_ENABLE_MPI)
```

Comment:
Let's simplify this logic with 
Total = POSIX + stdio
Ifdef hdf

Total += hdf
Endif 

And so on.

Reply:
Yeah, this makes sense 😄